app/services/api/charts/sector.rb

```diff
@@ -0,0 +1,148 @@
+          .sort.to_h
+          .map { |level, companies| [level, companies_summary(companies)] }
+      end
+
+      # Returns latest "MQ assessment Sector's levels" mapped to number of Companies in given level
+      #
+      # Returns single series of data.
+      #
+      # @example
+      #   { '0' => 13, '1' => 63, '2' => 61, '3' => 71, '4' => 63, '4STAR' => 6}
+      #
+      def companies_count
+        @company_scope
+          .includes(:mq_assessments)
+          .group_by { |company| company.mq_assessments.order(:assessment_date).first.level }
+          .map { |level, companies| [level, companies.size] }
+          .sort.to_h
+      end
+
+      # Returns companies emissions
+      #
+      # Returns data in multiple series.
+      #
+      # @example
+      #   [
+      #     { name: 'WizzAir', data: {} },
+      #     { name: 'Air China', data: {'2014' => 111.0, '2015' => 112.0 } },
+      #     { name: 'China Southern', data: {'2014' => 114.0, '2015' => 112.0 } }
+      #   ]
+      #
+      def companies_emissions_data
+        [
+          emissions_data_from_companies,
+          emissions_data_from_sector_benchmarks
+        ].flatten
+      end
+
+      # Returns Companies stats grouped by CP benchmark.
+      #
+      # Returns data in multiple series.
+      #
+      # @example
+      #   [
+      #     {
+      #       name: 'Below 2',
+      #       data: [ ['Coal Mining', 52], ['Steel', 73] ]
+      #     },
+      #     {
+      #       name: 'Paris',
+      #       data: [ ['Coal Mining', 65], ['Steel', 26] ]
+      #     }
+      #   ]
+      def group_by_cp_benchmark
+        CP_SCENARIOS.map do |cp_alignment|
+          {
+            name: get_alignment_label(cp_alignment),
+            data: companies_count_per_sector_cp_scenarios(cp_alignment)
+          }
+        end
+      end
+
+      private
+
+      def emissions_data_from_companies
+        @company_scope
+          .includes(:mq_assessments, :cp_assessments)
+          .map { |company| company_emissions_series_options(company) }
+      end
+
+      def emissions_data_from_sector_benchmarks
+        @company_scope.first.sector_benchmarks.map do |benchmark|
+          {
+            type: 'area',
+            fillOpacity: 0.1,
+            name: benchmark.scenario,
+            data: benchmark.emissions
+          }
+        end
+      end
+
+      def get_alignment_label(cp_alignment)
+        {
+          below_2: 'Below 2',
+          exact_2: '2 degrees',
+          paris: 'Paris',
+          not_aligned: 'Not aligned',
+          no_disclosure: 'No disclosure'
+        }[cp_alignment.to_sym]
+      end
+
+      def companies_count_per_sector_cp_scenarios(_cp_alignment)
+        ::Sector.pluck(:id, :name).map do |_sector_id, sector_name|
+          [sector_name, rand(100)]
+        end
+      end
+
+      def company_emissions_series_options(company)
+        {
+          name: company.name,
+          data: company.cp_assessments.last&.emissions,
```

Comment:
`last` could not be last here, as `has_many` scope is not ordered. We could add order by assessment_date directly to cp_assessments and mq_assessments relation's definitions. What do you think?

Reply:
yep, I think it's good idea to include this ordering in the scopes.

Reply:
a306e70

Reply:
Right, I've tried to fix it in separate PR as you've seen and came into the problem of code clarity ;], like `latest_assessment` looks better when it is `mq_assessment.last` so the default order should be not descending. On the other side in the UI we want descending order.

Reply:
Weirdo, I thought that Rails will reset relation order when I want to override it, like when I have order on relation and then want to reverse it like using `latest_first` that does not work. So, I think we should not add order on relation and keep the code as it is here https://github.com/Vizzuality/laws_and_pathways/pull/72

Reply:
ok, I moved the ordering from scope to method and added test case for it - 055dfa4.